API/Program.cs

```diff
@@ -71,6 +80,82 @@
         options.SubstituteApiVersionInUrl = true;
     });
 
+builder
+    .Services.AddRateLimiter(options =>
+    {
+        RateLimitConfiguration rateLimitConfiguration = builder.Configuration.BindAndValidate<RateLimitConfiguration>(
+            RateLimitConfiguration.Position);
+
+        var defaultFixedOptions = new FixedWindowRateLimiterOptions()
+        {
+            PermitLimit = 30,
+            Window = TimeSpan.FromSeconds(60),
+            QueueLimit = 0
+        };
+
+        // Configure response for rate limited clients
+        options.OnRejected = async (context, token) =>
+        {
+            context.HttpContext.Response.StatusCode = StatusCodes.Status429TooManyRequests;
+            const string readMore =
+                "Read more about our rate limits at https://github.com/osu-tournament-rating/otr-wiki/blob/master/api/limits/en.md";
+            if (context.Lease.TryGetMetadata(MetadataName.RetryAfter, out TimeSpan retryAfter))
+            {
+                await context.HttpContext.Response.WriteAsync(
+                    $"Too many requests. Please try again after {retryAfter.TotalSeconds} second(s). " + readMore,
+                    token
+                );
+            }
+            else
+            {
+                await context.HttpContext.Response.WriteAsync(
+                    "Too many requests. Please try again later. " + readMore, token);
+            }
+        };
+
+
+        options.GlobalLimiter = PartitionedRateLimiter.Create<HttpContext, string>(context =>
+        {
+            // Try to get access token
+            var accessToken =
+                context.Features.Get<IAuthenticateResultFeature>()?.AuthenticateResult?.Properties
+                    ?.GetTokenValue("access_token")?.ToString() ?? string.Empty;
+
+            // Shared partition for anonymous users
+            if (accessToken.IsNullOrEmpty())
+            {
+                return RateLimitPartition.GetFixedWindowLimiter(
+                    "anonymous",
+                    _ => defaultFixedOptions
+                );
+            }
+
+            // Try parse rate limit override claim
```

Comment:
```suggestion
            // Try to parse the 'ratelimitoverrides' claim
```

Reply:
Define a constant for the claim name ('ratelimitoverrides') in Utilities somewhere.

Reply:
Where would you include this? In something like a `Constants.cs` ?

Reply:
Nvm, it defo makes more sense to use an extension to the `ClaimTypes` class

Reply:
Actually, `System.Security.Claims.ClaimTypes` is a static class, so I went for a new `OtrClaimTypes` instead.